core/src/main/java/com/netflix/conductor/core/execution/mapper/ForkJoinDynamicTaskMapper.java

```diff
@@ -195,9 +195,17 @@ public List<TaskModel> getMappedTasks(TaskMapperContext taskMapperContext)
             }
 
             for (TaskModel forkedTask : forkedTasks) {
-                Map<String, Object> forkedTaskInput =
-                        tasksInput.get(forkedTask.getReferenceTaskName());
-                forkedTask.addInput(forkedTaskInput);
+                try {
+                    Map<String, Object> forkedTaskInput =
+                            tasksInput.get(forkedTask.getReferenceTaskName());
+                    forkedTask.addInput(forkedTaskInput);
+                } catch (Exception e) {
+                    String reason =
+                            String.format(
+                                    "Dynamic tasks could not be created due to invalid task input: %s ",
```

Comment:
nit: Since, `DYNAMIC` is a task type already, this would probably read better as ```Tasks could not be dynamically forked due to invalid input: %s```